crates/typst/src/model/numbering.rs

```diff
@@ -80,6 +80,19 @@ pub fn numbering(
     /// particularly useful in itself, it means that you can just give arbitrary
     /// numberings to the `numbering` function without caring whether they are
     /// defined as a pattern or function.
+    ///
+    /// Be careful when adding a suffix inside the function, as it will
+    /// introduce unwanted behavior when there is a punctuation (or any symbol)
+    /// after a reference to the element:
+    ///
+    /// ```example
+    /// #set heading(numbering: "1.")
+    /// = First <a>
+    /// This is a reference: @a.
+    /// #set heading(numbering: (..n) => numbering("1.", ..n))
+    /// == Second <b>
+    /// This is a reference: @b. Did you notice the second period?
+    /// ```
```

Comment:
I feel like we should first note that references trim prefixes and suffixes from typical numbering formats, and only then explain that this does not happen for functions.

Though this section isn't related to any change in 0.12, so this can be done in a separate PR if this discussion blocks anything.

Reply:
I dropped this for now in https://github.com/typst/typst/pull/5147. I don't really like documenting bugs, even though I guess it's sometimes pragmatic.